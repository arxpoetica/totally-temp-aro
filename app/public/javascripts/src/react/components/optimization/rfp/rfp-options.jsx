import React, { Component } from 'react'
import { Field, reduxForm } from 'redux-form'

export class RfpOptions extends Component {
  render () {
    return <div className='p-2 m-2'>
      <h4>Options</h4>
      <form className='d-flex flex-column rfp-options'
        style={{ height: '100%' }}
        onSubmit={event => event.preventDefault()}>
        <table className='table table-sm table-striped'>
          <tbody>
            {
              Object.keys(this.props.initialValues).map(optionKey => (
                <tr key={optionKey}>
                  <td>{this.props.initialValues[optionKey].displayName}</td>
                  <td>
                    <Field name={`${optionKey}.value`} className='form-control form-control-sm' component='select' type='text'>
                      <option value='ROUTE_FROM_NODES'>Route from nodes</option>
                      <option value='ROUTE_FROM_FIBER'>Route from fiber</option>
                    </Field>
                  </td>
                </tr>
              ))
            }
          </tbody>
        </table>
      </form>
    </div>
  }

  validateName (value) {
    value = value || ''
    return (value.indexOf(' ') >= 0) ? 'Name cannot have spaces' : undefined
  }
}

let RfpOptionsForm = reduxForm({
  form: 'rfp'
})(RfpOptions)

export default RfpOptionsForm
