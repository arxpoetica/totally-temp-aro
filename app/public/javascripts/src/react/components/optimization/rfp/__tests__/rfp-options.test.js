/* global test expect */
import React from 'react'
import { shallow } from 'enzyme'
import { RfpOptions } from '../rfp-options'

const options = {
  fiberRoutingMode: {
    displayName: 'Fiber routing mode',
    value: 'ROUTE_FROM_FIBER'
  }
}

// -----------------------------------------------------------------------------
test('Default options render', () => {
  const component = shallow(
    <RfpOptions initialValues={options} />
  )
  expect(component).toMatchSnapshot()
})
