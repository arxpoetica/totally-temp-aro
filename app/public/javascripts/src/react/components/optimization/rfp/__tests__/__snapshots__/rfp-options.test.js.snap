// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default options render 1`] = `
<div
  className="p-2 m-2"
>
  <h4>
    Options
  </h4>
  <form
    className="d-flex flex-column rfp-options"
    onSubmit={[Function]}
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <table
      className="table table-sm table-striped"
    >
      <tbody>
        <tr
          key="fiberRoutingMode"
        >
          <td>
            Fiber routing mode
          </td>
          <td>
            <Field
              className="form-control form-control-sm"
              component="select"
              id="field_fiberRoutingMode"
              name="fiberRoutingMode.value"
              type="text"
            >
              <option
                value="ROUTE_FROM_NODES"
              >
                Route from nodes
              </option>
              <option
                value="ROUTE_FROM_FIBER"
              >
                Route from fiber
              </option>
            </Field>
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</div>
`;
